framework/src/operators/packet_batch.rs

```diff
@@ -132,13 +132,15 @@ impl PacketBatch {
                 idx_new += 1;
             }
 
-            // We did not find an index that was passed in, warn/error out.
+            // We did not find an index that was passed in:
+            // Clear the batch
             if remove_idx < idxes.len() {
-                None
+                self.clear_packets();
```

Comment:
not sure this is right. if one of the indices passed in is not found we will drop the entire batch?